src/components/timeline/TimelineContextMenu.svelte

```diff
@@ -110,6 +111,18 @@
     const duration = includeDuration ? getIntervalInMs(span.duration) : 0;
     return new Date(getUnixEpochTimeFromInterval(startYmd, span.start_offset) + duration);
   }
+
+  function onZoom(duration: number) {
+    if (xScaleView && contextMenu) {
+      const time = xScaleView.invert(contextMenu.e.offsetX);
+      const newViewTimeRange: TimeRange = { end: time.getTime() + duration + duration, start: time.getTime() };
```

Comment:
My fault as I didn't finish this initial implementation but was imagining we'd want something like - center on the context menu time and enforce min max timeline bounds?